File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack bowl = new Stack();
        
        for(int move : moves){
            for(int i = 0; i < board.length; i++){
                if(board[i][move - 1] != 0){
                    if(bowl.empty()){
                        bowl.push(board[i][move - 1]);
                    }else{
                        if((int)bowl.peek() == board[i][move - 1]){
                            bowl.pop();
                            answer += 2;
                        }else bowl.push(board[i][move - 1]);
                    }
                    board[i][move - 1] = 0;
                    break;
                }
            }
        }
        
        return answer;
    }
    
}

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    var stack  = [];
    
    for(var i = 0; i < moves.length; i++){
        var now = moves[i]-1
        for(var j = 0;j < board.length; j++){
            if(board[j][now] != 0){
                if(stack[stack.length-1] === board[j][now]){
                    stack.pop();
                    answer += 2;
                }
                else{
                    stack.push(board[j][now])
                }
                board[j][now] = 0;
                break;
            }
        }
    }
    return answer
}